cfme/tests/infrastructure/test_snapshot.py

```diff
@@ -41,40 +43,52 @@ def domain(request):
     return dom
 
 
-@pytest.fixture(scope="module")
-def test_vm(setup_provider_modscope, provider, vm_name, request):
-    """Fixture to provision appliance to the provider being tested if necessary"""
-    vm = VM.factory(vm_name, provider, template_name=provider.data['full_template']['name'])
+def provision_vm(provider, vm_name, request, small_template=True):
+    if small_template:
+        vm = VM.factory(vm_name, provider, template_name=provider.data['small_template'])
+    else:
+        vm = VM.factory(vm_name, provider, template_name=provider.data['full_template']['name'])
 
     if not provider.mgmt.does_vm_exist(vm_name):
         vm.create_on_provider(find_in_cfme=True, allow_skip="default")
         request.addfinalizer(vm.delete_from_provider)
     return vm
 
 
-def new_snapshot(test_vm, has_name=True):
+@pytest.fixture(scope="module")
+def small_test_vm(setup_provider_modscope, provider, vm_name, request):
+    return provision_vm(provider, vm_name(), request)
+
+
+@pytest.fixture(scope="module")
+def full_test_vm(setup_provider_modscope, provider, vm_name, request):
+    return provision_vm(provider, vm_name(), request, small_template=False)
+
+
+def new_snapshot(test_vm, has_name=True, memory=False):
     if has_name:
         new_snapshot = Vm.Snapshot(
             name="snpshot_" + fauxfactory.gen_alphanumeric(8),
-            description="snapshot", memory=False, parent_vm=test_vm
+            description="snapshot", memory=memory, parent_vm=test_vm
         )
     else:
         new_snapshot = Vm.Snapshot(
             description="snapshot_" + fauxfactory.gen_alphanumeric(8),
-            memory=False, parent_vm=test_vm
+            memory=memory, parent_vm=test_vm
         )
     return new_snapshot
 
 
 @pytest.mark.uncollectif(lambda provider:
     not provider.one_of(RHEVMProvider, VMwareProvider) or
     (provider.one_of(RHEVMProvider) and provider.version < 4))
-def test_snapshot_crud(test_vm, provider):
+def test_snapshot_crud(small_test_vm, provider):
     """Tests snapshot crud
 
     Metadata:
         test_flag: snapshot, provision
     """
+    test_vm = small_test_vm
```

Comment:
Although I personally think this is a) faster and b) nicer, I did expect that I  would have to refactor :) On it.